src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1276,14 +1281,17 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 	case SN_CreateScalarUnsafe:
 		return emit_simd_ins_for_sig (cfg, klass, OP_CREATE_SCALAR_UNSAFE, -1, arg0_type, fsig, args);
 	case SN_Dot: {
-#ifdef TARGET_ARM64
+#if defined(TARGET_ARM64) || defined(TARGET_WASM)
 		if (!is_element_type_primitive (fsig->params [0]))
 			return NULL;
 
 		int instc0 = type_enum_is_float (arg0_type) ? OP_FMUL : OP_IMUL;
 		MonoInst *pairwise_multiply = emit_simd_ins_for_sig (cfg, klass, OP_XBINOP, instc0, arg0_type, fsig, args);
-
+#if defined(TARGET_ARM64)
 		return emit_sum_vector (cfg, fsig->params [0], arg0_type, pairwise_multiply);
+#else
+		return emit_sum_vector (cfg, klass, fsig, arg0_type, &pairwise_multiply);
```

Comment:
Shouldn't at least the signature be unified ?

Reply:
I used the same signature as amd64. I see that it changed meanwhile, so I will update it to use the same as arm64 and now amd64 as well.